pretext/ExtraTopics/None.ptx

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0"?>
+<section xml:id="extratopics-none">
+  <title>None</title>
+
+  <p><c>None</c>, or the <term>NoneType</term>, is a type which represents the <em>absence</em> of a value.
+    But let's take a step back and think: why is this useful?</p>
+
+  <p>In Python, sometimes we don't want to return anything from a function. However, when we don't 
+    explicitly return anything from a function, Python will implicitly return None.
+    You could also explicitly return None from a function <em>or</em> return a value,
+    and then you would have to check if the function returned something or not.</p>
+
+  <p>For the purposes of this class, you should just understand when None appears and why.</p>
+
+  <p>Most commonly when students are learning about strings and lists, us educators see a similar issue, which links back
+    to sections on <term>mutability</term>. Strings are immutable and lists are mutable, remember?</p>
+
+  <p>When we want to modify a string, we must reassign it, because it's immutable:</p>
+  <program language="python">
+    <input>
+my_string = 'Hello, world!'
+new_string = my_string[0:5]
+print(new_string)
+    </input>
+  </program>
+
+  <p>Your natural instinct may be to write the same thing with lists:</p>
+  <program language="python">
+    <input>
+my_list = []
+my_list = my_list.append(0)
+print(my_list[0])
+    </input>
+  </program>
+
+  <p>By doing so, you will incur this error:</p>
+  <program language="python">
+    <input>
+print(my_list[0])
+~~~~~~~^^^
+TypeError: 'NoneType' object is not subscriptable
+    </input>
+  </program>
+
+  <p>Frustratingly, your code actually did exactly what you told it to. List functions modify the list directly, because
+    lists are mutable. Therefore, their functions (such as <c>.append(element)</c>) don't return anything.</p>
+
+  <p>But now we know why: <c>append</c> is actually implicitly returning <c>None</c>. This means that in the previous example, we reassigned <c>my_list</c>
+    to <c>None</c>, and then we tried to list access element 0 of <c>None</c>, which caused the 'not subscriptable' error, because None isn't a list
+    and can't have list access done to it.</p>
+
+  <p>The solution is simple: don't reassign the list when calling list functions.</p>
+  <program language="python">
+    <input>
+my_list = []
+my_list.append(0)
+print(my_list[0])
+    </input>
+  </program>
+
+  <p>If you remember nothing else, remember: <term>strings get reassigned, lists don't</term>. This is because
+    strings are <term>immutable</term>; lists are mutable and their functions don't return anything (they implicitly return None!).</p>
```

Comment:
I regularly get questions on how to tell the difference between functions that return something or not. This is mentioning a good rule of thumb, but I might also add that the name of the method can often be a hint. We should probably also say they can check the documentation whenever they aren't sure. Sometimes you just need to look it up. I do that often enough myself still.